frontend/TestRun/Components/ResultTable.svelte

```diff
@@ -83,14 +106,86 @@
     function toggleDescription() {
         showDescription = !showDescription;
     }
+
+    async function fetchGraphData() {
+        try {
+            const params = queryString.stringify({
+                testId: test_id,
+                startDate,
+                endDate,
+                "tableNames[]": [table_name],
+            });
+
+            let res = await fetch(`/api/v1/test-results?${params}`);
+            if (res.status !== 200) {
+                throw new Error(`HTTP Error ${res.status} trying to fetch test results`);
+            }
+            let results = await res.json();
+            if (results.status !== "ok") {
+                throw new Error(`API Error: ${results.message}`);
+            }
+
+            const response = results["response"];
+            if (!response.graphs || response.graphs.length === 0) {
+                throw new Error("No graph data available for this table");
+            }
+
+            graph = response["graphs"].find((graph) => graph.options.plugins.title.text.endsWith(selectedColumnName));
+            ticks = response["ticks"];
+            releasesFilters = Object.fromEntries(response["releases_filters"].map((key) => [key, true]));
+        } catch (error) {
+            console.log("Error:", error);
+            sendMessage("error", `Failed to load graph data: ${error.message}`);
+        }
+    }
+
+    async function openGraphModal(columnName = "") {
+        selectedColumnName = columnName;
+        showGraphModal = true;
+        await fetchGraphData();
+    }
+
+    function closeGraphModal() {
+        showGraphModal = false;
+        graph = null;
+        selectedColumnName = "";
+    }
+
+    async function handleDateChange(event) {
+        startDate = event.detail.startDate;
+        endDate = event.detail.endDate;
+        if (showGraphModal) {
+            await fetchGraphData();
+            redraw++;
+        }
+    }
+
+    async function handleReleaseChange(event) {
+        releasesFilters = event.detail.releasesFilters;
+        redraw++;
+    }
+
+    function handleKeydown(event) {
+        if (event.key === "Escape" && showGraphModal) {
+            closeGraphModal();
+        }
+    }
```

Comment:
This function should probably be moved into `ModalWindow.svelte` component itself probably (and just dispatch closeEvent)

Reply:
moved